### PR TITLE
Us31 drag and drop

### PR DESCRIPTION
**Pull request by majluni**

### CODE REVIEW

File: src/gui/dragAndDrop.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ function dropNode(event) {
         clone.style.top = `${event.pageY}px`;
         clone.style.left = `${event.pageX}px`;
         event.target.appendChild(clone);
+        NodeForest.insertRootNode(nodeCount);
```
Comment (JonathanBush) [inline src/gui/dragAndDrop.js]: Adding nodes to the forest is already implemented by Us4.

File: src/gui/dragAndDrop.js [hunk 2]
```diff
@@ -35,16 +37,14 @@ function dropNode(event) {
...
 
     Calculator.updateResult();
-  // Add node to tree/forest here
 }
 
 function deleteNode(event) {
     event.preventDefault();
     const id = event.dataTransfer.getData("text");
     document.getElementById(id).remove();
+    NodeForest.deleteRootNode(nodeCount);
```
Comment (JonathanBush) [inline src/gui/dragAndDrop.js]: deleteRootNode should use the node ID, which should be equal to the HTML ID. This ID is generated when Node is instantiated and returned by NodeFactory.createNode().

**Review by JonathanBush — CHANGES_REQUESTED** [review]
Implementation conflicts with changes in #14 Us4 node parsing.

File: src/gui/dragAndDrop.js (hunk 1 above)
Comment (majluni) [reply src/gui/dragAndDrop.js]: I have merged this with mine, have another look

**Comment by JonathanBush** [comment]
Looks good. Proper use node id.

**Review by JonathanBush — APPROVED** [review]

**Pull request merged**